codegen/helper.go

```diff
@@ -0,0 +1,159 @@
+package codegen
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/iancoleman/strcase"
+	"github.com/streamingfast/eth-go"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strconv"
+)
+
+type CodegenEvent struct {
+	RustName string
+	Fields   map[string]string
+}
+
+func GetContractAbi(contract string) ([]byte, error) {
+	res, err := http.Get(fmt.Sprintf("https://api.etherscan.io/api?module=contract&action=getabi&address=%s&apikey=7E11P1IJ4ZRWQ36CZN78QZ7KTE7YP7MJ31", contract))
```

Comment:
The actual API key used should probably ce coming from a flag and have the default value. Also, where this API key was generated exactly? And can we use it without one?

Reply:
it can be done without an API key, changed.